### DISCUSSION TITLE
discord.slash namespace

### DISCUSSION THREAD

**Discussion by esmaycat** [Pycord-Development/pycord#19] [Q&A]
Just wondering what anyone's thoughts are on putting everything slash-command related into a `discord.slash` namespace like `discord.ui` that already exists for buttons + select menus. 

At the moment everything lives in the `discord/bot.py`, perhaps it would be better to structure it like so 
```
discord/
  slash/
    bot.py
    options.py
    commands.py
  # everything else here
```
With this approach you would have to do 
```py
from discord.slash import x
# or just...
from discord import slash
slash.x
```
Not sure where the context-menu related stuff would live, perhaps it could go in here as well?

**Comment by HarryPotter12352** [comment]
The context class should be in the slash/ directory in my opinion. That way we would be using `discord.slash.SlashContext` which is suitable

**Reply by esmaycat** [reply]
Instead of passing Context (or a form of it) to the slash commands, we could just pass the `discord.Interaction` object that already exists. From there you can do `interaction.response.send_message` etc...

**Reply by NonImportantPerson** [reply]
Yea that would be better but I still think it should be in discord.Slash

**Comment by CodeWithSwastik** [comment]
We already did! It's now under `discord/app`

**Reply by CodeWithSwastik** [reply]
It's soon being changed to `discord/commands` as that makes more sense

**Comment by Dorukyum** [comment]
The namespace for application commands has been set to `discord.commands`.
